src/EIP712.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {IERC5267} from "openzeppelin-contracts/contracts/interfaces/IERC5267.sol";
+import {IEIP712} from "./interfaces/IEIP712.sol";
+
+/// @title EIP712
+///
+/// @dev This contract does not cache the domain separator and calculates it on the fly since it will change when delegated to.
+/// @notice It is not compatible with use by proxy contracts since the domain name and version are cached on deployment.
+///
+/// @author Uniswap
+/// @author Modified from Coinbase (https://github.com/coinbase/smart-wallet)
+abstract contract EIP712 is IEIP712, IERC5267 {
```

Comment:
is there a reason this needs to be abstract?

Reply:
Not really, can change